sources/platform/actors/development/actor_definition/dataset_schema/validation.md

```diff
@@ -0,0 +1,230 @@
+
+:::
+
+You can either do that directly through `actor.json`:
+
+```json title=".actor.json"
+{
+    "actorSpecification": 1,
+    "storages": {
+        "dataset": {
+            "actorSpecification": 1,
+            "fields": {
+                "$schema": "http://json-schema.org/draft-07/schema#",
+                "type": "object",
+                "properties": {
+                    "name": {
+                        "type": "string"
+                    }
+                },
+                "required": ["name"]
+            },
+            "views": {}
+        }
+    }
+}
+```
+
+Or in a separate separate file linked from the `.actor.json`:
+
+```json title=".actor.json"
+{
+    "actorSpecification": 1,
+    "storages": {
+        "dataset": "./dataset_schema.json"
+    }
+}
+```
+
+```json title="dataset_schema.json"
+{
+    "actorSpecification": 1,
+    "fields": {
+        "$schema": "http://json-schema.org/draft-07/schema#",
+        "type": "object",
+        "properties": {
+            "name": {
+                "type": "string"
+            }
+        },
+        "required": ["name"]
+    },
+    "views": {}
+}
+```
+
+:::important
+
+Dataset schema needs to be a valid JSON schema draft-07, so the `$schema` line is important and must be exactly this value or it must be omitted:
+
+`"$schema": "http://json-schema.org/draft-07/schema#"`
+
+:::
+
+## Dataset validation
+
+When you define a schema of your default dataset, the schema is then always used when you insert data into the dataset to perform validation (we use [AJV](https://ajv.js.org/)).
+
+If the validation succeeds, nothing changes from the current behavior, data is stored and an empty response with status code 201 is returned.
+
+**If the data you attempt to store in the dataset is invalid** (meaning any of the items received by the API fails the validation), **the whole request is discarded** and the API will return a response with status code 400 and the following JSON response:
+
+```json
+{
+    "error": {
+        "type": "schema-validation-error",
+        "message": "Schema validation failed",
+        "data": {
+            "invalidItems": [{
+                "itemPosition": "<array index in the received array of items>",
+                "validationErrors": "<Complete list of AJV validation error objects>"
+            }]
+        }
+    }
+}
+```
+
+The type of the AJV validation error object is [here](https://github.com/ajv-validator/ajv/blob/master/lib/types/index.ts#L86)
+
+If you use the Apify JS client or Apify SDK and call `pushData` function you can access the validation errors in a `try catch` block like this:
+
+```javascript
+try {
+    const response = await Actor.pushData(items);
+} catch (error) {
+    if (!error.data?.invalidItems) throw error;
+    error.data.invalidItems.forEach((item) => {
+        const { itemPosition, validationErrors } = item;
+    });
+}
+```
```

Comment:
What about Python Client & SDK? If we provide code sample for one, it would make sense to provide it for the other as well, and utilize [Tabs component](https://docusaurus.io/docs/markdown-features/tabs)